packages/ragbits-evaluate/src/ragbits/evaluate/dataset_generator/utils.py

```diff
@@ -0,0 +1,15 @@
+import re
+from difflib import SequenceMatcher
+from itertools import combinations
+
+def get_closest_substring(long, short):
+    a, b = max(
+        combinations(re.finditer('|'.join(short.split()), long), 2),
+        key=lambda c: SequenceMatcher(None, long[c[0].start():c[1].end()], short).ratio()
+    )
+    return long[a.start():b.end()]
+
+def get_passages_list(raw_passages: str) -> list[str]:
+    passages = raw_passages.split("[")[1]
```

Comment:
what if llm is mistaken about intended format ? or the raw passage lists looks lie that:

"['increment x by one' , sample random number from [x, x+1, x+2]']" ?